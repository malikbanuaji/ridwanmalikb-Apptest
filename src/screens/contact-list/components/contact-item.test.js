import * as React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import ContactItem from './contact-item';

jest.mock('../../../components');

describe('ContactItem', () => {
  const props = {
    onPress: jest.fn(),
    onLongPress: jest.fn(),
    profilePicture: 'N/A',
    firstName: 'Malik',
    lastName: 'Banuaji',
    id: '1',
    index: 0,
  };

  test('should render properly without profile', () => {
    const component = render(<ContactItem {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render properly with profile', () => {
    const component = render(
      <ContactItem {...props} profilePicture="file://uri" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should handle onPress', () => {
    const component = render(
      <ContactItem {...props} profilePicture="file://uri" />,
    );
    const button = component.getByTestId('ContactItem');
    fireEvent(button, 'press');
    expect(props.onPress).toBeCalled();
  });
});
