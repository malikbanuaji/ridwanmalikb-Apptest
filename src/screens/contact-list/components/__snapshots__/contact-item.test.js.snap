// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem should render properly with profile 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.2)",
        "borderRadius": 4,
        "marginBottom": 5,
        "marginHorizontal": 10,
        "overflow": "hidden",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="ContactItem"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "height": 42,
            "marginRight": 10,
            "overflow": "hidden",
            "width": 42,
          }
        }
      >
        <ProfilePicture
          photoUri="file://uri"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "fontWeight": "400",
              "letterSpacing": 0.2,
            }
          }
        >
          Malik
           
          Banuaji
        </Text>
      </View>
      <View
        style={
          Object {
            "marginRight": 5,
          }
        }
      >
        <AntDesignIcon
          color="#ffffff"
          name="right"
          size={14}
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`ContactItem should render properly without profile 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.2)",
        "borderRadius": 4,
        "marginBottom": 5,
        "marginHorizontal": 10,
        "overflow": "hidden",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="ContactItem"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "height": 42,
            "marginRight": 10,
            "overflow": "hidden",
            "width": 42,
          }
        }
      >
        <ProfileText
          firstName="Malik"
          fontSize={18}
          lastName="Banuaji"
          size={42}
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "fontWeight": "400",
              "letterSpacing": 0.2,
            }
          }
        >
          Malik
           
          Banuaji
        </Text>
      </View>
      <View
        style={
          Object {
            "marginRight": 5,
          }
        }
      >
        <AntDesignIcon
          color="#ffffff"
          name="right"
          size={14}
        />
      </View>
    </View>
  </View>
</View>
`;
